Extensions/TextInput/textinputruntimeobject.ts

```diff
@@ -0,0 +1,377 @@
+namespace gdjs {
+  const logger = new gdjs.Logger('Text input object');
+
+  const supportedInputTypes = [
+    'text',
+    'email',
+    'password',
+    'number',
+    'telephone number',
+    'url',
+    'search',
+    'text area',
+  ] as const;
+
+  type SupportedInputType = typeof supportedInputTypes[number];
+
+  const parseInputType = (potentialInputType: string): SupportedInputType => {
+    const lowercasedNewInputType = potentialInputType.toLowerCase();
+
+    // @ts-ignore - we're actually checking that this value is correct.
+    if (supportedInputTypes.includes(lowercasedNewInputType))
+      return potentialInputType as SupportedInputType;
+
+    return 'text';
+  };
+
+  /** Base parameters for {@link gdjs.TextInputRuntimeObject} */
+  export type TextInputObjectDataType = {
+    /** The base parameters of the TextInput */
+    content: {
+      initialValue: string;
+      placeholder: string;
+      fontResourceName: string;
+      fontSize: float;
+      inputType: SupportedInputType;
+      textColor: string;
+      fillColor: string;
+      fillOpacity: float;
+      borderColor: string;
+      borderOpacity: float;
+    };
+  };
+  export type TextInputObjectData = ObjectData & TextInputObjectDataType;
+
+  const defaultWidth = 300;
```

Comment:
Since this is a "global" constant, I would recommend putting an all-caps name to distinguish it from local variables:  `DEFAULT_WIDTH`, ditto for the constant below.

Reply:
Done 👍